### PR TITLE
Find my safe

### PR DESCRIPTION
**Pull request by nicosampler**

### CODE REVIEW

File: src/apps/FindMySafe/index.tsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    try {
+      const res = await getBalances(safe, safeInfo!.network);
+      setBalances(res);
+      setLoadingBalances(false);
+    } catch (error) {
+      setLoadingBalances(false);
+    }
+  };
+
+  const getBalancesContent = () => {
```
Comment (mikheevm) [inline src/apps/FindMySafe/index.tsx:198]: this file is getting too big and hard to follow, is it possible to split it into separate components? for example, you can extract this to a separate component `Balances`

Comment (nicosampler) [reply src/apps/FindMySafe/index.tsx:198]: Sorry, wasn't ready to review yet. I created it to test something and I forget to put it on WIP

Comment (nicosampler) [reply src/apps/FindMySafe/index.tsx]: @mikheevm ready for review now!

**Comment by rmeissner** [comment]
May I ask why this should be a safe app? (You never use the safe apps functionality here)

**Comment by mikheevm** [comment]
I agree with @rmeissner, there's a lot of requests of users trying to find their safe address. This app, as I understand, aims to help, but in order to use it, one has to create a new safe, which is weird imo

**Comment by nicosampler** [comment]
yes, guys, you are right, I missed that use case and seems to be the most important. I was thinking about users that already have a safe. Luckily this is pretty simple to move to the safe, let's ask @lukasschor his opinion.

**Comment by nicosampler** [comment]
It works as a standalone app as well. You can check it here https://upbeat-kowalevski-f8a155.netlify.app/find-my-safe

**Comment by lukasschor** [comment]
I agree with Mikhail & Richard. And we have a ticket for introducing a feature natively to automatically suggest Safes to the user where he/she is an owner of: https://github.com/gnosis/safe-react/issues/510

However, I still think it's good that we have this Safe App because of a few reasons:
- The nice thing about Safe Apps is that we don’t have to touch safe-react code
- Less overhead (incl. design, etc.) and easier to implement
- QA does not have to check if the feature potentially broke anything somewhere else
- It works as a standalone app that we can already refer to from the Help Center or support channels
- It also showcases that there are potential Safe Apps that don't necessarily need to be "Transaction crafter" apps

**Comment by rmeissner** [comment]
For the showcase usecase I find the erc721 collection a better example. Also as @nicosampler a standalone website makes more sense for this.

Also I would like to know the purpose of this repo? Should it be for examples or for default apps developed by gnosis? There is a difference in maintenance.

**Comment by nicosampler** [comment]
@rmeissner sure, could you create a repo?

**Comment by rmeissner** [comment]
https://github.com/gnosis/safe-finder (we can rename it if required, but I would like to have something that starts with `safe` ;) )

**Comment by mikheevm** [comment]
@rmeissner What do you think about adding it to the interface itself? Maybe on the load screen we can include a button "find my safe", or on the welcome screen. I don't think it's very hard

**Comment by rmeissner** [comment]
I don'T really care, this is a decision that needs to be made by @lukasschor

**Comment by lukasschor** [comment]
I would propose to put this on hold for now, and think about how to integrate this as a native feature instead.

**Comment by rmeissner** [comment]
I would still close this PR in this case. @nicosampler feel free to move the code to the new repo ;)
